app/jobs/attach_files_to_work_job.rb

```diff
@@ -2,43 +2,60 @@
 class AttachFilesToWorkJob < ActiveJob::Base
   queue_as Hyrax.config.ingest_queue_name
 
+  before_enqueue do |job|
+    log = job.arguments.last
+    log.pending_job(self)
+  end
+
   # @param [ActiveFedora::Base] the work class
   # @param [Array<UploadedFile>] an array of files to attach
-  def perform(work, uploaded_files)
+  # @param [Hyrax::Operation] a log storing the status of the job
+  def perform(work, uploaded_files, log)
+    log.performing!
     uploaded_files.each do |uploaded_file|
       file_set = FileSet.new
       user = User.find_by_user_key(work.depositor)
       actor = Hyrax::Actors::FileSetActor.new(file_set, user)
       actor.create_metadata(visibility: work.visibility)
-      attach_content(actor, uploaded_file.file)
+      attach_content!(actor, uploaded_file.file, log)
       actor.attach_file_to_work(work)
       actor.file_set.permissions_attributes = work.permissions.map(&:to_hash)
 
       uploaded_file.update(file_set_uri: file_set.uri)
     end
+    # TODO: Should this depend on whether child_log was a success?
```

Comment:
When the child log is finalized, it checks to see if it's parent has any open children. If they are all finalized, it sets the parent as a final state too (success or failure).